### PR TITLE
Developer can style content types output differently per viewport

### PR DESCRIPTION
**Pull request by omiroshnichenko**
:heavy_exclamation_mark::heavy_exclamation_mark:DO NOT MERGE BEFORE https://github.com/magento/magento2-page-builder/issues/558 closed :heavy_exclamation_mark::heavy_exclamation_mark:
## Description
Convert style blocks to inline styles to prevent BIC changes

## Related Issue
https://jira.corp.magento.com/browse/PB-558
magento/magento2-page-builder#558

### Verification Steps

1. Go to any page with Page Builder content.
2. Verify that content renders as before.

## Checklist

- [ ] I have added tests to cover my changes, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/pagebuilder/lib/parseStorageHtml.js [hunk 1]
```diff
@@ -74,6 +74,48 @@ const walk = (rootEl, contentTypeStructureObj) => {
...
+
+    if (styleBlocks.length > 0) {
+        Array.from(styleBlocks).forEach(styleBlock => {
+            const cssRules = styleBlock.sheet.cssRules;
+
+            Array.from(cssRules).forEach(rule => {
+                const selectors = rule.selectorText
+                    .split(',')
+                    .map(selector => selector.trim());
+                selectors.forEach(selector => {
```
Comment (bluemwhitew) [inline packages/pagebuilder/lib/parseStorageHtml.js:96]: Readability: New line after `const` and/or before `forEach` loop.

File: packages/pagebuilder/lib/parseStorageHtml.js [hunk 2]
```diff
@@ -74,6 +74,48 @@ const walk = (rootEl, contentTypeStructureObj) => {
...
+                    if (!styles[selector]) {
+                        styles[selector] = [];
+                    }
+                    styles[selector].push(rule.style);
```
Comment (bluemwhitew) [inline packages/pagebuilder/lib/parseStorageHtml.js:100]: Readability: New line after `if` block.

**Review by bluemwhitew — APPROVED** [review]

**Comment by davemacaulay** [comment]
The code looks great, and I appreciate the solution builds in backwards compatibility with older versions of Page Builder. How do we intend to handle media queries once they're introduced for responsive aspects of Page Builder? We can't just push these back into inline styles.

Performance is of huge importance for PWA Studio, could you run a performance analysis of old content which has the style tags inline and then new content which needs to be converted to help better understand the negative impact this has on performance.

**Comment by omiroshnichenko** [comment]
@davemacaulay 
Performance results performed on the page which contains all content types.
Before: 
Screen Shot 2020-10-05 at 5 04 54 PM
After:
Screen Shot 2020-10-05 at 5 03 41 PM
Lighthouse results are the same.

Media queries and other styles could be stored in the registry and received to read in configAgregators. This is will be the next iteration of changes in the Page Builder PWA package.

**Comment by davemacaulay** [comment]
@omiroshnichenko looks good, the 2ms increase in time for the `parseStorageHtml` function is acceptable in my eyes.

**Review by davemacaulay — APPROVED** [review]

**Comment by bluemwhitew** [comment]
Added [Manual Test](https://www.github.com/magento/magento2-page-builder/issues/558#issuecomment-710091731) to parent issue magento/magento2-page-builder#558.

**Review by tjwiebell — APPROVED** [review]

**Comment by dpatil-magento** [comment]
QA Approved.

**Pull request merged**